File: cakk-domain/src/main/java/com/cakk/domain/config/DataSourceConfig.java
package com.cakk.domain.config;

import javax.sql.DataSource;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;

@Configuration
public class DataSourceConfig {

	@Bean
	@ConfigurationProperties(prefix = "storage.datasource.core")
	public HikariConfig hikariConfig() {
		return new HikariConfig();
	}

	@Bean
	@Primary
	public DataSource dataSource() {
		return new HikariDataSource(hikariConfig());
	}
}

File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/config/DataSourceConfig.kt
package com.cakk.domain.mysql.config

import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import javax.sql.DataSource

@Configuration
class DataSourceConfig {
    @Bean
    @ConfigurationProperties(prefix = "storage.datasource.core")
    fun hikariConfig(): HikariConfig {
        return HikariConfig()
    }

    @Bean
    @Primary
    fun dataSource(): DataSource {
        return HikariDataSource(hikariConfig())
    }
}
